### ISSUE TITLE
Fix alternate language metadata links

### ISSUE THREAD

**Issue by cmichi** [paritytech/ink-docs#211]
## The Issue

use.ink is available in English and Spanish. For SEO it's important to have a meta HTML tag on the English page that specifies what the corresponding page in the Spanish language code is. And vice-versa, the Spanish page needs a meta tag to specifywhat the English page is.

Docusaurus does this automatically, but it can't handle translated slugs. So e.g. for English we have:

```
slug: /why-rust-for-smart-contracts
```
For Spanish:

```
slug: /por-que-rust-para-smart-contracts
```

Docusaurus has a naive logic and just inserts this into each page:

```html
<head>
   …
   <link rel="alternate" href="https://use.ink/es/why-rust-for-smart-contracts" hreflang="es-ES" data-rh="true">
   …
</head>
```

So it uses the English slug and just prepends `/es/`. This is unfortunate, as that page does not exist. The link 404's. Hence Google currently does not link our English docs to the Spanish properly.

docusaurus does not not support i18n for slugs, this is clearly stated as a non-goal [here](https://docusaurus.io/docs/i18n/introduction#i18n-non-goals). There is an issue in that repo, but it won't happen soon.

## The Solution

We don't have too many pages in our docs, we can just annotate the corresponding English/Spanish page everywhere. This would look something like:

```md
---
title: Why Rust for Smart Contracts?
hide_title: true
slug: /why-rust-for-smart-contracts
slug.es-ES: /por-que-rust-para-smart-contracts
---
```

You would then have to modify how docusaurus output the `<link rel="alternate" href="https://use.ink/es/why-rust-for-smart-contracts" hreflang="es-ES" data-rh="true">` for each page.

## Approaches

A plugin would be most elegant, but I would also happily accept a pragmatic solution.

For example, docusaurus supports the concept of [swizzling](https://docusaurus.io/docs/swizzling). A possible approach would be to do 

```
yarn swizzle @docusaurus/theme-classic SiteMetadata  --eject
```

You will then have a file `src/theme/SiteMetadata/index.js` locally that contains the docusaurus logic that needs to be modified. This file would have to be upgraded with newer docusaurus versions as well, which is not ideal.

Swizzling also supports wrapping, not sure if you can find a pragmatic way of replacing e.g. each `/es/$english_slug` with `/es/$spanish_slug`.

Open for other approaches!

**Comment by peetzweg** [comment]
Just looked into it a bit more. It seems the only issue with the out of the box docusaurus solution is that it's not working with translated slugs as you mentioned. Is there even any benefit to this?

I would prefer not translating the slugs anyways and keep them all in english. Will be just unnecessary custom code which might break and complicates adaption more languages later on.

```
What we "want" to work
https://use.ink/why-rust-for-smart-contracts
https://use.ink/es/por-que-rust-para-smart-contracts


With not translated slugs it looks like this:
https://use.ink/why-rust-for-smart-contracts
https://use.ink/es/why-rust-for-smart-contracts
```

We have some pages with translated slug, these don't work with docusaurus out of the box
Example: `why-rust-for-smart-contracts` => `por-que-rust-para-smart-contracts`

Markdown files which not translated the slugs work with the docusaurus added header tag.
Example: `/getting-started/building-your-contract` => `/getting-started/building-your-contract`

So we already have in inconsistency inside the docs code base regarding slug translation. If we agree on not translating the slugs I will revert the translated slugs to english for this issues.

https://user-images.githubusercontent.com/839848/229727146-4b187426-3cb9-4c79-a523-2d53a42f6049.mov

**Comment by peetzweg** [comment]
tl;dr: SEO wants spanish urls. 🤦

**Comment by peetzweg** [comment]
### Using a Swizzled SiteMetadata

Pretty painful to implement this. Not as straightforward as expected as I haven't found a way yet to access the front matter / doc metadata data from the swizzled `SiteMetadata.js` component.

An approach to get this working is to just disable the code which creates the `hreflang` header entries. And add them manually in each doc using the <head> directly.

IMHO this does not feel maintainable and worthwhile at all.

https://github.com/paritytech/ink-docs/blob/794c6e27f42267a5d9d3898d274e0a7d8c51507a/src/theme/SiteMetadata/index.js#L13-L32

https://github.com/paritytech/ink-docs/blob/794c6e27f42267a5d9d3898d274e0a7d8c51507a/docs/intro/why-rust.md?plain=1#L1-L13


Tried all mentioned hooks in the Client API section:
https://docusaurus.io/docs/next/docusaurus-core#hooks

I've search the repos issue without success and contacted support on docusaurus discord. No response yet though.

### Using Plugin

Creating a plugin could be feasible but is quite complex. As the front matter seems to be handled by `plugin-content-docs`, we need to basically recreate this plugin and augment it.

Btw even if we than add the proper hreflang <link> tag to each doc programmatically, we probably need to "disable" it in a the "unsafe" swizzeld SiteMetadata.js .

### Lighthouse Test

All in all I'm not sure if this hassle is really worth it. I wasn't able to find proper seo best practices for "translated" urls but our lighthouse score does not seem to care about. Other links I've found for this topic also not argue to much for translating actual URLs.

https://webmasters.stackexchange.com/questions/85918/should-i-translate-url-for-better-seo

https://developers.google.com/search/docs/specialty/international/localized-versions

<img width="548" alt="SCR-20230404-ldkc" src="https://user-images.githubusercontent.com/839848/229765052-af74d9ac-af15-45b0-84d7-808801b158e2.png">
